plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/stock-filter/edit.tsx

```diff
@@ -1,120 +1,115 @@
+						},
+						style: {
+							spacing: {
+								blockGap: '0',
+							},
+						},
+					},
+					[
+						[
+							'core/heading',
+							{
+								level: 3,
+								content: __( 'Status', 'woocommerce' ),
+							},
+						],
+						[
+							'woocommerce/product-filter-clear-button',
+							{
+								lock: {
+									remove: true,
+									move: false,
+								},
+							},
+						],
+					],
+				],
+				[
+					'woocommerce/product-filter-chips',
+					{
+						lock: {
+							remove: true,
+						},
+					},
+				],
+			],
+		}
 	);
 
-	const { results: filteredCounts } = useCollectionData( {
+	const { results, isLoading } = useCollectionData( {
 		queryStock: true,
 		queryState: {},
 		isEditor: true,
 	} );
 
-	const listOptions = useMemo( () => {
-		return Object.entries( stockStatusOptions )
-			.map( ( [ key, value ] ) => {
-				const count = (
-					filteredCounts as unknown as CollectionData
-				 )?.stock_status_counts?.find(
-					( item: StockStatusCount ) => item.status === key
-				)?.count;
+	const stock = getStockFilterData( results );
 
-				return {
-					value: key,
-					label: (
-						<Label
-							name={ value }
-							count={
-								showCounts && count ? Number( count ) : null
-							}
-						/>
-					),
-					count: count || 0,
-				};
-			} )
-			.filter( ( item ) => item.count > 0 );
-	}, [ stockStatusOptions, filteredCounts, showCounts ] );
+	const labels = useMemo(
+		() => ( {
+			instock: __( 'In stock', 'woocommerce' ),
+			outofstock: __( 'Out of stock', 'woocommerce' ),
+			onbackorder: __( 'On backorder', 'woocommerce' ),
+		} ),
+		[]
+	);
+
+	const data = stock.map( ( { status, count } ) => {
+		const label = labels[ status ];
+		return {
+			label: label + ` (${ count.toString() })`,
+			value: status,
+		};
+	} );
 
 	return (
-		<>
-			{
-				<div { ...blockProps }>
-					<Inspector { ...props } />
-					<Disabled>
-						<div
-							className={ clsx( `style-${ displayStyle }`, {
-								'is-loading': false,
-							} ) }
-						>
-							{ displayStyle === 'dropdown' ? (
-								<>
-									<PreviewDropdown
-										placeholder={
-											props.attributes.selectType ===
-											'single'
-												? __(
-														'Select stock status',
-														'woocommerce'
-												  )
-												: __(
-														'Select stock statuses',
-														'woocommerce'
-												  )
-										}
-									/>
-								</>
-							) : (
-								<CheckboxList
-									className={ 'wc-block-stock-filter-list' }
-									options={ listOptions }
-									checked={ [] }
-									onChange={ () => {
-										// noop
-									} }
-									isLoading={ false }
-									isDisabled={ true }
-								/>
-							) }
-						</div>
-					</Disabled>
-				</div>
-			}
-		</>
+		<div { ...innerBlocksProps }>
+			<InitialDisabled>
+				<BlockContextProvider
+					value={ {
+						filterData: {
+							items: data,
+							stock,
```

Comment:
This isn't used in the inner blocks.

Reply:
Fixed with 1dcb2cc94d355685a911efb813c9ef9465b69030